From 53a077ec6dc161f1c538c42cfa733aa1bb309d6b Mon Sep 17 00:00:00 2001
From: Harish Rachakonda <quic_rachakon@quicinc.com>
Date: Tue, 19 Nov 2024 11:35:45 +0530
Subject: [PATCH] UPSTREAM:CP ath12k: Added support for 256 clients

Currently, ath12k supports 128 clients. Support has been added for
256 clients for the 1G profile and 128 clients for the 512M profiles.

Patch-dependency: 835-ath12k-Add-512MB-support-changes.patch
patch-work: None

Signed-off-by: Harish Rachakonda <quic_rachakon@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/hw.h | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/hw.h b/drivers/net/wireless/ath/ath12k/hw.h
index 5812d79..fd49bb1 100644
--- a/drivers/net/wireless/ath/ath12k/hw.h
+++ b/drivers/net/wireless/ath/ath12k/hw.h
@@ -20,6 +20,10 @@ struct ath12k_cfr_peer_tx_param;
 #ifdef CPTCFG_ATH12K_MEM_PROFILE_512M
 /* Num VDEVS per radio */
 #define TARGET_NUM_VDEVS	(8 + 1)
+
+/* Max num of stations (per radio) */
+#define TARGET_NUM_STATIONS	128
+
 /* Num of Bridge vdevs per radio */
 #define TARGET_NUM_BRIDGE_VDEVS	0
 #define ATH12K_MAX_NUM_VDEVS_NLINK	TARGET_NUM_BRIDGE_VDEVS
@@ -32,6 +36,9 @@ struct ath12k_cfr_peer_tx_param;
 #define ATH12K_MAX_NUM_VDEVS_NLINK	TARGET_NUM_VDEVS + \
 					TARGET_NUM_BRIDGE_VDEVS
 #define ATH12K_QMI_TARGET_MEM_MODE	ATH12K_QMI_TARGET_MEM_MODE_DEFAULT
+
+/* Max num of stations (per radio) */
+#define TARGET_NUM_STATIONS	256
 #endif
 
 /* ToDo: do we need to include self peer of bridge vdev also here? */
@@ -46,9 +53,6 @@ struct ath12k_cfr_peer_tx_param;
 /* Num of peers for DBS_SBS */
 #define TARGET_NUM_PEERS_DBS_SBS	(3 * TARGET_NUM_PEERS_PDEV)
 
-/* Max num of stations (per radio) */
-#define TARGET_NUM_STATIONS	128
-
 #define TARGET_NUM_PEERS(x)	TARGET_NUM_PEERS_##x
 #define TARGET_NUM_PEER_KEYS	2
 /* Do we need to change the below */
-- 
2.34.1

